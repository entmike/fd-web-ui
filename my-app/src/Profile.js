import React from "react";
import { useAuth0 } from "@auth0/auth0-react";
import {Avatar} from '@chakra-ui/react';

const Profile = () => {
  const { user, isAuthenticated, isLoading } = useAuth0();

  if (isLoading) {
    return <div>Loading ...</div>;
  }

  return (
    isAuthenticated && (
      <Avatar
        size={'sm'}
        src={user.picture} alt={user.name}
      />
    )
  );
};

export default Profile;