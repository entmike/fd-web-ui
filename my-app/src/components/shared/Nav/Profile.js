import React from 'react';
import { useAuth0 } from '@auth0/auth0-react';
import { Avatar } from '@chakra-ui/react';

export function Profile() {
  const { user, isAuthenticated, isLoading } = useAuth0();

  if (isLoading) {
    return <div>Loading ...</div>;
  }

  return (
    isAuthenticated && (
      <Avatar
        aria-label="user-avatar"
        size={'sm'}
        src={user.picture}
        alt={user.name}
      />
    )
  );
}
