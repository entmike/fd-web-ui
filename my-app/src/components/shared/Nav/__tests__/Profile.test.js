import React from 'react';
import { Profile } from '../Profile';
import { render, screen } from '@testing-library/react';
import { useAuth0 } from '@auth0/auth0-react';

jest.mock('@auth0/auth0-react');

describe('Profile test suite', () => {
  const component = <Profile />;
  const setup = () => render(component);

  beforeEach(() => {
    useAuth0.mockReturnValue({
      user: {},
      isAuthenticated: false,
      isLoading: false,
    });
  });

  afterEach(jest.clearAllMocks);

  it('should generate a snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  describe('when is not loading', () => {
    beforeEach(() => {
      useAuth0.mockReturnValue({
        user: { picture: 'test', name: 'test' },
        isAuthenticated: true,
        isLoading: false,
      });
    });

    it('should render avatar', () => {
			setup()
      expect(screen.getByLabelText('user-avatar')).toBeInTheDocument();
    });
  });

  describe('when is loading', () => {
    beforeEach(() => {
      useAuth0.mockReturnValue({
        user: {},
        isAuthenticated: false,
        isLoading: true,
      });
    });
    it('should render loading message', () => {
			setup()
      expect(screen.getByText('Loading ...')).toBeInTheDocument();
    });
  });
});
