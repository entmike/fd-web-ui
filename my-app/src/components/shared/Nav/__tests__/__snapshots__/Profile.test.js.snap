// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile test suite should generate a snapshot 1`] = `<div />`;
